httpserver/service_node.cpp

```diff
@@ -200,7 +203,8 @@ ServiceNode::ServiceNode(boost::asio::io_context& ioc, uint16_t port,
 
     BOOST_LOG_TRIVIAL(info) << "Requesting initial swarm state";
     swarm_timer_tick();
-    pow_difficulty_timer_tick();
+    pow_dns_thread_ = std::make_unique<boost::thread>(boost::bind(&ServiceNode::pow_difficulty_timer_tick, this));
+    pow_dns_thread_->detach();
```

Comment:
Unless I'm missing something, the `pow_dns_thread_` will execute `pow_difficulty_timer_tick` once then it will stop. The subsequent calls will still be executed in the main thread.
The reason is that `pow_difficulty_timer_tick` re-schedules itself via `pow_update_timer_.async_wait()` which is executed in the main thread by the `ioc`.
We need to find a way to have another the `pow_update_timer_` attached to another `ioc` which is executed in the new thread...

Reply:
And it would be nicer to not detach the thread, and instead do a graceful stop (e.g. stop the ioc and join the thread) upon deconstruction of `service_node`.

Reply:
True, seems like we need a separate io_context in another thread.

Reply:
Updated to use a second io_context